tests/test_symbolic_primitives.py

```diff
@@ -250,45 +265,56 @@ def test_wipe(self, env, prim_gen, sponge, sink, countertop):
         mud_system = env.scene.get_system("mud")
         countertop.states[object_states.Covered].set_value(mud_system, True)
-        assert countertop.states[object_states.Covered].get_value(mud_system)
 
-        # First toggle on the sink
-        for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.TOGGLE_ON, sink):
-            env.step(action)
-        assert sink.states[object_states.ToggledOn].get_value()
-
-        # Then grasp the sponge
+        # First grasp the sponge
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.GRASP, sponge):
             env.step(action)
         assert robot.states[object_states.IsGrasping].get_value(sponge)
 
+        # Then toggle on the sink
+        sink.states[object_states.ToggledOn].set_value(True)
+        assert sink.states[object_states.ToggledOn].get_value()
+
         # Then soak the sponge under the water
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.SOAK_UNDER, sink):
             env.step(action)
         assert sponge.states[object_states.Saturated].get_value(water_system)
 
+        # Then toggle on the sink
+        sink.states[object_states.ToggledOn].set_value(False)
+        assert not sink.states[object_states.ToggledOn].get_value()
+
         # Wipe the countertop with the sponge
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.WIPE, countertop):
             env.step(action)
         assert not countertop.states[object_states.Covered].get_value(mud_system)
 
-    @pytest.mark.skip(reason="primitives are broken")
     def test_cut(self, env, prim_gen, apple, knife, countertop):
+        # Store the apple object information for scene reset
+        deleted_objs = [apple]
+        deleted_objs_cfg = [retrieve_obj_cfg(obj) for obj in deleted_objs]
+
         # assert not apple.states[object_states.Cut].get_value(knife)
+        # start a new environment to enable transition rules
         print("Grasping knife")
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.GRASP, knife):
             env.step(action)
-        for _ in range(60):
-            env.step(prim_gen._empty_action())
         print("Cutting apple")
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.CUT, apple):
             env.step(action)
-        for _ in range(60):
-            env.step(prim_gen._empty_action())
         print("Putting knife back on countertop")
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.PLACE_ON_TOP, countertop):
             env.step(action)
 
+        # clean up
+        half_apples = env.scene.object_registry("category", "half_apple", set()).copy()
+        for apple in half_apples:
+            env.scene.remove_object(apple)
+
+        objs = [DatasetObject(**obj_cfg) for obj_cfg in deleted_objs_cfg]
+        og.sim.batch_add_objects(objs, scenes=[env.scene] * len(objs))
+        og.sim.step()
```

Comment:
Again, your cleanup relies on the test passing. This is not good. You can save the state after the initial load and then do load_state in the env fixture function right before every test. Lets discuss if you are unsure how to implement this.

Reply:
Recommendation: remove this

Reply:
lets just do this: go ahead and remove the manual reset and add the pytest skip decorator to this test with a message saying "Disabled until env reset can add/remove objects" (edited)